docs/abi.rst

```diff
@@ -0,0 +1,560 @@
+
+.. code-block:: python
+
+    @Subroutine(TealType.uint64)
+    def minimum(a: abi.Uint64, b: abi.Uint64) -> Expr:
+        """Return the minimum value of the two arguments."""
+        return (
+            If(a.get() < b.get())
+            .Then(a.get())
+            .Else(b.get())
+        )
+
+Getting Values at Indexes
+''''''''''''''''''''''''''
+
+The types :code:`abi.StaticArray`, :code:`abi.Address`, :code:`abi.DynamicArray`, :code:`abi.String`, and :code:`abi.Tuple` are compound types, meaning they contain other types whose values can be extracted. The :code:`__getitem__` method, accessible by using square brackets to "index into" an object, can be used to extract these values.
+
+The supported methods are:
+
+* :any:`abi.StaticArray.__getitem__`, used for :code:`abi.StaticArray` and :code:`abi.Address`
+* :any:`abi.Array.__getitem__`, used for :code:`abi.DynamicArray` and :code:`abi.String`
+* :any:`abi.Tuple.__getitem__`
+
+Be aware that these methods return a :code:`ComputedValue`, TODO link to Computed Value section
+
+A brief example is below. Please consult the documentation linked above for each method to learn more about specific usage and behavior.
+
+.. code-block:: python
+
+    @Subroutine(TealType.none)
+    def ensure_all_values_greater_than_5(array: abi.StaticArray[abi.Uint64, L[10]]) -> Expr:
+        """This subroutine asserts that every value in the input array is greater than 5."""
+        i = ScratchVar(TealType.uint64)
+        return For(
+            i.store(Int(0)), i.load() < array.length(), i.store(i.load() + Int(1))
+        ).Do(
+            array[i.load()].use(lambda value: Assert(value.get() > Int(5)))
+        )
+
+Limitations
+"""""""""""""""""""""
+
+TODO: explain type size limitations
+
+Reference Types
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+Some AVM operations require specific values to be placed in the "foreign arrays" of the app call transaction. Reference types allow methods to describe these requirements.
+
+Reference types are only valid in the arguments of a method. They may not appear in a method's return value in any form.
+
+Definitions
+""""""""""""""""""""""""""""""""""""""""""
+
+PyTeal supports the following reference types:
+
+====================== ====================== ================ =======================================================================================================================================================
+PyTeal Type            ARC-4 Type             Dynamic / Static Description
+====================== ====================== ================ =======================================================================================================================================================
+:any:`abi.Account`     :code:`account`        Static           Represents an additional account that the current transaction can access, stored in the :any:`Txn.accounts <TxnObject.accounts>` array
```

Comment:
I think the word "additional" is potentially confusing here - additional to what? If you mean in addition to the app's account, then maybe it makes sense, but then I would at least remove it from the assets line.